oemof/solph/custom.py

```diff
@@ -839,3 +844,582 @@ def tes_ub_constr_rule(block, n, t):
             return (self.tes_level[n, t] <= n.params['tes_level_max'])
         self.tes_ub_constr = Constraint(
             self.GENERICCAES, m.TIMESTEPS, rule=tes_ub_constr_rule)
+
+
+########################################################################
+
+# ----------------------- DSM Component --------------------------------
+class SinkDSM(Sink):
+    r"""
+    Module for creating a Demand Side Management component.
+
+    Based on the paper by Zerrahn, Alexander and Schill, Wolf-Peter (2015):
+    On the representation of demand-side management in power system models,
+    in: Energy (84), pp. 840-845, 10.1016/j.energy.2015.03.037,
+    accessed 17.09.2019, pp. 842-843.
+
+    A special sink component which modifies the input demand series.
+
+    Parameters
+    ----------
+    demand: int or array
```

Comment:
I assume, floats are allowed. We use to write "numeric (sequence or scalar)" in cases like these.